### PR TITLE
[WIP] Prefer standard `where` over AREL

### PR DESCRIPTION
**Pull request by bdunne**

### CODE REVIEW

File: lib/insights/api/common/filter.rb [hunk 1]
```diff
@@ -111,55 +111,55 @@ def integer(k, val)
...
-          self.query = query.where(key => value)
+          self.query = query.where(key => Array(value).collect { |i| ActiveRecord::Base.sanitize_sql(i) })
         end
 
         def comparator_eq_i(key, value)
-          values = Array(value).map { |v| query.sanitize_sql_like(v.downcase) }
+          values = Array(value).collect { |i| ActiveRecord::Base.sanitize_sql(i.downcase) }
 
-          self.query = query.where(arel_table[key].lower.matches_any(values))
+          self.query = query.where("lower(#{key}) IN (?)", values)
```
Comment (carbonin) [inline lib/insights/api/common/filter.rb]: If we're interpolating the key here, should we use something like this to sanitize it? https://www.rubydoc.info/gems/pg/PG/Connection#quote_ident-class_method

(this applies generally to this PR)

File: lib/insights/api/common/filter.rb [hunk 2]
```diff
@@ -111,55 +111,55 @@ def integer(k, val)
...
         end
 
         def comparator_gt(key, value)
-          self.query = query.where(arel_table[key].gt(value))
+          self.query = query.where("#{key} > ?", value)
```
Comment (carbonin) [inline lib/insights/api/common/filter.rb]: Found this which explains an interesting side-effect of this change we might want to consider https://stackoverflow.com/a/35186805

Specifically that we lose the table qualification here so if we're joining across tables that have the same key as a column we will run into errors.

File: lib/insights/api/common/filter.rb (hunk 1 above)
Comment (bdunne) [reply lib/insights/api/common/filter.rb]: The only things that ever get this far are documented attributes.  Do you think we still need to quote them?

File: lib/insights/api/common/filter.rb (hunk 2 above)
Comment (bdunne) [reply lib/insights/api/common/filter.rb]: interesting

**Comment by bdunne** [comment]
It doesn't look like this is going to work with the tag filtering I need to do

**Comment by carbonin** [comment]
So just as a recap, did this fix the issue with the acts_as_tenant gem?
Also curious what about the filtering would make a difference between these implementations.

**Pull request closed without merging**